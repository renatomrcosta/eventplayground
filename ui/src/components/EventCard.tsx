import styles from "./EventCard.module.scss";
import React from "react";
import {
  Button,
  Typography,
  Container,
  Card,
  CardContent,
  CardActions,
} from "@material-ui/core";

export interface Props {
  title: string;
  description: string;
  onClick?: () => void;
}

export const EventCard: React.FC<Props> = ({ onClick, title, description }) => {
  return (
    <Card variant="outlined">
      <CardContent>
        <Typography variant="h2" gutterBottom>
          {title}
        </Typography>
        <Typography variant="body1" gutterBottom>
          {description}
        </Typography>
      </CardContent>
      <CardActions className={styles.action}>
        <Button
          variant="contained"
          color="primary"
          onClick={() => onClick && onClick()}
        >
          Im interested!
        </Button>
      </CardActions>
    </Card>
  );
};
