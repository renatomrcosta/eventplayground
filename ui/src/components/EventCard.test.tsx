import React from "react";
import { render, fireEvent, screen } from "@testing-library/react";
import { EventCard, Props } from "./EventCard";

describe("Event Card Tests", () => {
  const defaultProps: Props = {
    title: "Some title",
    description: "Some description",
  };
  it("Should render nicely", () => {
    const wrapper = render(<EventCard {...defaultProps} />);
    expect(wrapper).toMatchSnapshot();
  });

  it("should fire an event on button click", () => {
    //Given
    const clickHandler = jest.fn();
    render(<EventCard {...defaultProps} onClick={clickHandler} />);

    //Then
    fireEvent.click(screen.getByText("Im interested!"));

    //Expect
    expect(clickHandler).toBeCalled();
  });
});
