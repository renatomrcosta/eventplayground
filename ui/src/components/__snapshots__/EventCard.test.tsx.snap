// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Event Card Tests Should render nicely 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="MuiPaper-root MuiCard-root MuiPaper-outlined MuiPaper-rounded"
      >
        <div
          class="MuiCardContent-root"
        >
          <h2
            class="MuiTypography-root MuiTypography-h2 MuiTypography-gutterBottom"
          >
            Some title
          </h2>
          <p
            class="MuiTypography-root MuiTypography-body1 MuiTypography-gutterBottom"
          >
            Some description
          </p>
        </div>
        <div
          class="MuiCardActions-root action MuiCardActions-spacing"
        >
          <button
            class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary"
            tabindex="0"
            type="button"
          >
            <span
              class="MuiButton-label"
            >
              Im interested!
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </button>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="MuiPaper-root MuiCard-root MuiPaper-outlined MuiPaper-rounded"
    >
      <div
        class="MuiCardContent-root"
      >
        <h2
          class="MuiTypography-root MuiTypography-h2 MuiTypography-gutterBottom"
        >
          Some title
        </h2>
        <p
          class="MuiTypography-root MuiTypography-body1 MuiTypography-gutterBottom"
        >
          Some description
        </p>
      </div>
      <div
        class="MuiCardActions-root action MuiCardActions-spacing"
      >
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary"
          tabindex="0"
          type="button"
        >
          <span
            class="MuiButton-label"
          >
            Im interested!
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </button>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
